src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -1288,50 +768,1631 @@ TEST_F(TrajectoryPlannerTest, velocity_trajectory_curved_path_low_acceleration)
+                        path_parameters.max_allowable_linear_speed);
+        EXPECT_FLOAT_EQ(trajectory.trajectory_elements[i].angular_speed,
+                        path_parameters.max_allowable_angular_speed);
+    }
+
+    status = app_trajectory_planner_modifyTrajectoryToBackwardsContinuous(
+        &trajectory, segment_lengths);
+    EXPECT_EQ(status, OK);
+    status = app_trajectory_planner_modifyTrajectoryToBackwardsContinuous(
+        &trajectory, segment_lengths);
+    EXPECT_EQ(status, OK);
+
+    // Check that the decceleration period is correct
+    // Check that the values of the acceleration period are correct
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[path_parameters.num_segments - 5].linear_speed,
+        1 - 0, 0.0001);
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[path_parameters.num_segments - 4].linear_speed,
+        0.9709, 0.0001);
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[path_parameters.num_segments - 3].linear_speed,
+        0.7928, 0.0001);
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[path_parameters.num_segments - 2].linear_speed,
+        0.5606, 0.0001);
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[path_parameters.num_segments - 1].linear_speed,
+        path_parameters.final_linear_speed, 0.0001);
+
+
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[path_parameters.num_segments - 5].angular_speed,
+        0.9428, 0.0001);
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[path_parameters.num_segments - 4].angular_speed,
+        0.8165, 0.0001);
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[path_parameters.num_segments - 3].angular_speed,
+        0.6666, 0.0001);
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[path_parameters.num_segments - 2].angular_speed,
+        0.4714, 0.0001);
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[path_parameters.num_segments - 1].angular_speed, 0,
+        0.0001);
+}
+
+TEST_F(TrajectoryPlannerTest, test_generate_time_profile_no_angular_profile)
+{
+    // Create a scenario where the linear path parameters will limit the speed of the
+    // trajectory because there is no orientation path to follow
+    FirmwareRobotPathParameters_t path_parameters;
+    path_parameters.max_allowable_angular_acceleration = 100;
+    path_parameters.max_allowable_angular_speed        = 100;
+    path_parameters.max_allowable_linear_acceleration  = 1;
+    path_parameters.max_allowable_linear_speed         = 1;
+    path_parameters.num_segments                       = 100;
+    path_parameters.initial_linear_speed               = 1;
+    path_parameters.final_linear_speed                 = 1;
+    path_parameters.t_start                            = 1;
+    path_parameters.t_end                              = 2;
+    path_parameters.path                = {.x = {0, 0, 1, 0}, .y = {0, 0, 1, 0}};
+    path_parameters.orientation_profile = {.coefficients = {0, 0, 0, 0}};
```

Comment:
It seems like there is an orientation profile here, but this test seems to say that it doesn't have one? I think I'm misunderstanding something here.

Reply:
Outdated.